inductor: fix compile error of bfloat16 broadcast operation

XiaobingSuper opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #104319



For the bfloat16 broadcast, there is always has compile error:
```
error: could not convert ‘tmp2’ from ‘Vectorized<float>’ to ‘Vectorized<c10::BFloat16>
```

This PR will fix this issue.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78

XiaobingSuper committed:
inductor: fix compile error fot bfloat16 broadcast operation

[ghstack-poisoned]

jgong5 commented on torch/_inductor/codegen/cpp.py:
```diff
@@ -1346,7 +1346,7 @@ def load(self, name: str, index: sympy.Expr):
+        if is_broadcast and dtype not in [torch.bfloat16]:
             if is_mask:
                 loadbuf = f"flag_to_float_scalar({loadbuf})"
             line = f"at::vec::Vectorized<float>(static_cast<float>({loadbuf}))"
```
Is it better to enhance this line to incorporate the bfloat16 broadcast instead of adding extra checks in the `else` path?

XiaobingSuper replied:
Changed.

XiaobingSuper committed:
Update on "inductor: fix compile error of bfloat16 broadcast operation"




For the bfloat16 broadcast, there is always has compile error:
```
error: could not convert ‘tmp2’ from ‘Vectorized<float>’ to ‘Vectorized<c10::BFloat16>
```

This PR will fix this issue.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78

[ghstack-poisoned]

XiaobingSuper committed:
Update on "inductor: fix compile error of bfloat16 broadcast operation"




For the bfloat16 broadcast, there is always has compile error:
```
error: could not convert ‘tmp2’ from ‘Vectorized<float>’ to ‘Vectorized<c10::BFloat16>
```

This PR will fix this issue.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78

[ghstack-poisoned]

XiaobingSuper committed:
Update on "inductor: fix compile error of bfloat16 broadcast operation"




For the bfloat16 broadcast, there is always has compile error:
```
error: could not convert ‘tmp2’ from ‘Vectorized<float>’ to ‘Vectorized<c10::BFloat16>
```

This PR will fix this issue.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78

[ghstack-poisoned]

XiaobingSuper committed:
Update on "inductor: fix compile error of bfloat16 broadcast operation"




For the bfloat16 broadcast, there is always has compile error:
```
error: could not convert ‘tmp2’ from ‘Vectorized<float>’ to ‘Vectorized<c10::BFloat16>
```

This PR will fix this issue.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78

[ghstack-poisoned]

XiaobingSuper commented:
@pytorchbot merge
